main_test.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/elastic/hey-apm/conv"
+	"github.com/elastic/hey-apm/strcoll"
+	"github.com/stretchr/testify/assert"
+)
+
+// All JSON fields in Input are required, with zero values being meaningful.
+func TestDefaultInput(t *testing.T) {
+	os.Args[1] = "-bench"
+	expectedZeroValues := []string{
+		"transaction_generation_limit",
+		"transaction_generation_frequency",
+		"spans_generated_max_limit",
+		"spans_generated_min_limit",
+		"error_generation_limit",
+		"error_generation_frequency",
+		"error_generation_frames_max_limit",
+		"error_generation_frames_min_limit",
+	}
+
+	input := parseFlags()
+	assert.True(t, input.IsBenchmark)
+	for k, v := range conv.ToMap(input) {
+		if strcoll.Contains(k, expectedZeroValues) {
+			continue
```

Comment:
This test implies it tests zero values for the flag `-bench`, which it doesn't really. 

The testcase would also succeed if the flag is not set, as it doesn't do any check for the `expectedZeroValues`. Not sure what exactly you are testing here but you should either run this for both - benchmark and non benchmark - if you want to test no additional flags are allowed to be zero, or add something like: 
```
if strcoll.Contains(k, expectedZeroValues) {
  assert.Zero(t, v)
  continue
}
```
if you want to test zero values for benchmark flag.

Reply:
Not sure I understand the feedback. This test does check the `bench` flag, and wouldn't pass otherwise: https://github.com/elastic/hey-apm/pull/134/files#diff-53618cae32e23ae47e1c35f10da92654R29

This is testing that fields can't have the zero value unless explicitly allowed in the test, so as to ensure that #132 doesn't happen again when adding new fields. I don't care if the fields in the exception list are 0 by default or not, just that they can be 0.

I am ok with adding a test for non benchmark as well (`bench` false wouldn't trigger the bug)

Reply:
got it, it is only a naming confusion then: `expectedZeroValues` implied to me that you want to test somewhere that those values are zero, rather than just excluding those keys from being tested for non zero. 

Btw testing for non zero should be doable with `assert.NotZero(t, v)`.